### PR TITLE
Have sandbox/sweep-reads2 output FASTQ records when quality is present

### PR DESCRIPTION
**Pull request by standage**
Because.

### CODE REVIEW

**Comment by standage** [comment]
Whaddya think @ctb?

**Comment by standage** [comment]
Bump @ctb.

File: sandbox/sweep-reads2.py [hunk 1]
```diff
@@ -88,7 +92,11 @@ def main():
         count = ht.get_median_count(record.sequence)[0]
         if count:
             m += 1
-            outfp.write('>%s\n%s\n' % (record.name, record.sequence))
+            if hasattr(record, 'quality'):
```
Comment (ctb) [inline sandbox/sweep-reads2.py]: see khmer.utils.write_record :)

**Comment by ctb** [comment]
Might be worth switching to using broken_paired_reader as well, so pairs of sequences stay together...

**Comment by camillescott** [comment]
Also, refer to #1484: there are problems with assuming existence of accuracy attribute signifying FASTQ

**Comment by standage** [comment]
Bump @ctb.

File: sandbox/sweep-reads2.py [hunk 2]
```diff
@@ -75,21 +80,25 @@ def main():
...
 
-        count = ht.get_median_count(record.sequence)[0]
-        if count:
-            m += 1
-            outfp.write('>%s\n%s\n' % (record.name, record.sequence))
-        n += 1
+        if is_pair:
+            count1 = ht.get_median_count(read1.sequence)[0]
+            count2 = ht.get_median_count(read2.sequence)[0]
+            if count1 and count2:
```
Comment (ctb) [inline sandbox/sweep-reads2.py]: I would go with 'or' here.

File: sandbox/sweep-reads2.py [hunk 3]
```diff
@@ -75,21 +80,25 @@ def main():
...
+                m += 1
+                write_record(read1, outfp)
```
Comment (ctb) [inline sandbox/sweep-reads2.py]: write_record_pair, I think.

**Comment by standage** [comment]
Done.

**Comment by ctb** [comment]
LGTM! I kinda like the broken_paired_reader interface I have to say.

**Comment by standage** [comment]
@ctb Can we merge these branches into master within the next couple of days? I've had to maintain duplicates of `sweep-reads2.py` in HPC so that I can run scripts from other branches.

**Comment by ctb** [comment]
Sure; the changes were approved 10 days ago...?

**Comment by standage** [comment]
Two things.

- I guess I'm still carrying baggage from when GitHub didn't have an explicit review process, and the rule was to have someone else merge your PRs. I figured the new "Approve" feature put the onus back on the submitter (or at least gives them permission) to merge their own PRs, but I'm not sure I ever saw any explicit discussion on this.
- This PR merged into a non-master branch, so my question remains. Shall I initiate a PR against master?

Thanks! :-)

**Comment by ctb** [comment]
On Mon, Nov 28, 2016 at 05:15:22PM -0800, Daniel Standage wrote:
> Two things.
> 
> - I guess I'm still carrying baggage from when GitHub didn't have an explicit review process, and the rule was to have someone else merge your PRs. I figured the new "Approve" feature put the onus back on the submitter (or at least gives them permission) to merge their own PRs, but I'm not sure I ever saw any explicit discussion on this.

fair enough

> - This PR merged into a non-master branch, so my question remains. Shall I initiate a PR against master?

oh! sure.

**Pull request merged**